scene/resources/packed_scene.cpp

```diff
@@ -124,6 +124,40 @@ Ref<Resource> SceneState::get_remap_resource(const Ref<Resource> &p_resource, Ha
 	return remap_resource;
 }
 
+static void _parse_dictionary_hint_string(const String &p_hint_string, Variant::Type &r_key_subtype, PropertyHint &r_key_subtype_hint, String *r_key_class_name, Variant::Type &r_value_subtype, PropertyHint &r_value_subtype_hint, String *r_value_class_name) {
+	r_key_subtype = Variant::NIL;
+	r_key_subtype_hint = PropertyHint::PROPERTY_HINT_NONE;
+	r_value_subtype = Variant::NIL;
+	r_value_subtype_hint = PropertyHint::PROPERTY_HINT_NONE;
+
+	int key_value_separator = p_hint_string.find_char(';');
+	if (key_value_separator >= 0) {
+		int key_subtype_separator = p_hint_string.find_char(':');
+		String key_subtype_string = p_hint_string.substr(0, key_subtype_separator);
+		int key_slash_pos = key_subtype_string.find_char('/');
+		if (key_slash_pos >= 0) {
+			r_key_subtype_hint = PropertyHint(key_subtype_string.get_slice("/", 1).to_int());
+			key_subtype_string = key_subtype_string.substr(0, key_slash_pos);
+		}
+		r_key_subtype = Variant::Type(key_subtype_string.to_int());
+		if (r_key_class_name != nullptr) {
+			*r_key_class_name = p_hint_string.substr(key_subtype_separator + 1, key_value_separator - (key_subtype_separator + 1));
```

Comment:
This doesn't work for `Dictionary<MyEnum, Node>` (see https://github.com/godotengine/godot/issues/101801). It will take the enum hint_string `Value1,Value2,Value3` as the `class_name` which results in the following error when calling `Dictionary::set_typed`:

```
ERROR: Class names can only be set for type OBJECT.
```

I don't think you can assume that the hint_string will always be the class_name. I think that only works when the `PropertyHint` is `PROPERTY_HINT_NODE_TYPE` or `PROPERTY_HINT_RESOURCE_TYPE`.